KeyDistribution_BB84/Tasks.qs

```diff
@@ -104,106 +105,113 @@ namespace Quantum.Kata.KeyDistribution {
+    operation T26_BB84Protocol_Test () : Unit {
+        // 1. Alice chooses a random set of bits to encode in her qubits 
+        //    and a random set of bases to prepare her qubits in.
         // ...
-        return new Bool[0];
+
+        // 2. Alice allocates qubits, encodes them using her choices and sends them to Bob.
+        //    (Note that you can not reflect "sending the qubits to Bob" in Q#)
+        // ...
+
+        // 3. Bob chooses a random set of bases to measure Alice's qubits in.
+        // ...
+
+        // 4. Bob measures Alice's qubits in his chosen bases.
+        // ...
+
+        // 5. Alice and Bob compare their chosen bases and use the bits in the matching positions to create a shared key.
+        // ...
+
+        // 6. Alice and Bob check to make sure nobody eavesdropped by comparing a subset of their keys
+        //    and verifying that more than a certain percentage of the bits match.
+        // For this step, you can check the percentage of matching bits using the entire key 
+        // (in practice only a subset of indices is chosen to minimize the number of discarded bits).
+        // ...
+
+        // If you've done everything correctly, the generated keys will always match, since there is no eavesdropping going on.
+        // In the next section you will explore the effects introduced by eavesdropping.
     }
 
 
     //////////////////////////////////////////////////////////////////
     // Part III. Eavesdropping
     //////////////////////////////////////////////////////////////////
 
-    // Task 3.1.
-    // Implement an eavesdropper, Eve. Eve will intercept a qubit, q from 
-    // the quantum channel that Alice and Bob are using. She will measure 
-    // in the basis defined in the basis parameter, basis, (false for rectangular and 
-    // true for diagonal), reconstruct the qubit into the original state q, 
-    // and return her measurement. Eve hopes that if she properly 
-    // reconstructs the qubit after measurement she won't be caught!
-    operation Task31 (q : Qubit, basis : Bool) : Int {
+    // Task 3.1. Eavesdrop!
+    // In this task you will try to implement an eavesdropper, Eve. 
+    // Eve will intercept a qubit from the quantum channel that Alice and Bob are using. 
+    // She will measure it in either the |0⟩/|1⟩ basis or the |+⟩/|-⟩ basis,
+    // reconstruct the qubit into the original state and send it back to the channel. 
+    // Eve hopes that if she properly reconstructs the qubit after measurement she won't be caught!
+    //
+    // Inputs:
+    //      1) "q" - a qubit in one of the following states: |0⟩, |1⟩, |+⟩ or |-⟩,
+    //      2) "basis" - Eve's guess of the basis she should use for measuring.
+    // Output: the bit encoded in the qubit (false for |0⟩/|+⟩ states, true for |1⟩/|-⟩ states).
+    // Note: in this task you are guaranteed that the basis you're given matches 
+    //       the one in which the qubit is encoded.
```

Comment:
Hi @tcNickolas 

I did not understand what was this Note referring to. Further, I think it will be useful to have the definition of basis in the task description, as was in the original comment on the left:

> basis, (false for rectangular and true for diagonal),

On a side note: thanks so much for maintaining these Katas, they are extremely useful to understand quantum computing concepts in small chunks. Also, thanks to @hwerbel for this well designed kata. 😄

Reply:
Thank you for pointing this out!! I agree, we shouldn't make the learner to scroll back to recall what the basis agreement is. Adding a note `"(false indicates |0⟩/|1⟩ basis, and true indicates |+⟩/|-⟩ basis)"` in the description of the input would be very helpful.

The note means that (unlike the real eavesdropping scenario) you are given the correct basis in which the qubit is encoded (so if you're given states |0⟩/|1⟩, you'll be given `basis = false`, you don't need to worry about the mismatch of the qubit state you're given and the basis). It should probably be extended to add `"so if you're given states |0⟩/|1⟩, you'll be given basis = false, and if you're given states |+⟩/|-⟩, you'll be given basis = true"`.

Do you want to try and create a PR to fix the task description? (I'm not sure if you're participating in [Hacktoberfest](https://hacktoberfest.digitalocean.com/), if yes, this would be a good entry for it!) If not, I'll get to fixing it at some point this week.

P.S. Thank you, I'm really glad you find the Katas helpful!

Reply:
Sure, I will send a PR in a couple of hours. Thanks for the detailed responses.